leshan-server-core/src/main/java/org/eclipse/leshan/server/security/SecurityInfo.java

```diff
@@ -65,7 +71,15 @@ public static SecurityInfo newPreSharedKeyInfo(String endpoint, String identity,
      */
     public static SecurityInfo newRawPublicKeyInfo(String endpoint, PublicKey rawPublicKey) {
         Validate.notNull(rawPublicKey);
-        return new SecurityInfo(endpoint, null, null, rawPublicKey);
+        return new SecurityInfo(endpoint, null, null, rawPublicKey, null);
+    }
+
+    /**
+     * Construct a {@link SecurityInfo} when using DTLS with an X509 Certificate.
+     */
+    public static SecurityInfo newX509CertInfo(String endpoint, X509Certificate[] X509CertChain) {
```

Comment:
Is it needed to provide a certificate chain?
I think that we should call something like securityInfo.isX509() in the RegistrationHandler.
Maybe we could have an enum in the SecurityInfo to get the type of security?